packages/react/src/components/Tree/TreeItem.tsx

```diff
@@ -0,0 +1,236 @@
+  id: string
+
+  /** The index of the item among its siblings. Count starts at 1. */
+  index?: number
+
+  /** Array of props for sub tree. */
+  items?: ShorthandCollection<TreeItemProps>
+
+  /** Level of the tree/subtree that contains this item. */
+  level?: number
+
+  /** Called when a tree title is clicked. */
+  onTitleClick?: ComponentEventHandler<TreeItemProps>
+
+  /** Called when the item's first child is about to be focused. */
+  onFocusFirstChild?: ComponentEventHandler<TreeItemProps>
+
+  /** Called when the item's siblings are about to be expanded. */
+  onSiblingsExpand?: ComponentEventHandler<TreeItemProps>
+
+  /** Called when the item's parent is about to be focused. */
+  onFocusParent?: ComponentEventHandler<TreeItemProps>
+
+  /** Whether or not the item is in the open state. Only makes sense if item has children items. */
+  open?: boolean
+
+  /** The id of the parent tree item, if any. */
+  parent?: ShorthandValue<TreeItemProps>
+
+  /** Array with the ids of the tree item's siblings, if any. */
+  siblings?: ShorthandCollection<TreeItemProps>
+
+  /**
+   * A custom render iterator for rendering each tree title.
+   * The default component, props, and children are available for each tree title.
+   *
+   * @param {React.ReactType} Component - The computed component for this slot.
+   * @param {object} props - The computed props for this slot.
+   * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+   */
+  renderItemTitle?: ShorthandRenderFunction
+
+  /** Properties for TreeTitle. */
+  title?: ShorthandValue<TreeTitleProps>
+}
+
+export interface TreeItemState {
+  treeSize: number // size of the tree without children.
+  hasSubtree: boolean
+}
+
+class TreeItem extends UIComponent<WithAsProp<TreeItemProps>, TreeItemState> {
+  static create: Function
+
+  static displayName = 'TreeItem'
+
+  static className = 'ui-tree__item'
+
+  static slotClassNames: TreeItemSlotClassNames = {
+    title: `${TreeItem.className}__title`,
+    subtree: `${TreeItem.className}__subtree`,
+  }
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      content: false,
+    }),
+    id: PropTypes.string.isRequired,
+    index: PropTypes.number,
+    items: customPropTypes.collectionShorthand,
+    level: PropTypes.number,
+    onTitleClick: PropTypes.func,
+    onFocusFirstChild: PropTypes.func,
+    onFocusParent: PropTypes.func,
+    onSiblingsExpand: PropTypes.func,
+    open: PropTypes.bool,
+    parent: customPropTypes.itemShorthand,
+    renderItemTitle: PropTypes.func,
+    siblings: customPropTypes.collectionShorthand,
+    title: customPropTypes.itemShorthand,
+    treeItemRtlAttributes: PropTypes.func,
+  }
+
+  static defaultProps = {
+    as: 'div',
+    accessibility: treeItemBehavior,
+  }
+
+  state = {
+    hasSubtree: false,
+    treeSize: 0,
+  }
+
+  static getDerivedStateFromProps(props: TreeItemProps) {
+    return {
+      hasSubtree: Tree.hasSubtree(props),
```

Comment:
```
ERROR in packages/react/dist/es/components/Tree/Tree.js -> packages/react/dist/es/components/Tree/TreeItem.js -> packages/react/dist/es/components/Tree/Tree.js
```

This create a circular reference, I suggest to create a simple function as there are no context requirements and move it to `Tree/lib/hasSubtree.ts`:

 ```suggestion
      hasSubtree: hasSubtree(props),
```